pandas/core/computation/expr.py

```diff
@@ -813,18 +819,27 @@ class Expr:
     parser : str, optional, default 'pandas'
     env : Scope, optional, default None
     truediv : bool, optional, default True
-    level : int, optional, default 2
+    level : int, optional, default 0
     """
 
     def __init__(
-        self, expr, engine="numexpr", parser="pandas", env=None, truediv=True, level=0
+        self,
+        expr,
+        engine: str = "numexpr",
+        parser: str = "pandas",
+        env=None,
+        truediv: bool = True,
+        level: int = 0,
     ):
         self.expr = expr
         self.env = env or Scope(level=level + 1)
         self.engine = engine
         self.parser = parser
         self.env.scope["truediv"] = truediv
-        self._visitor = _parsers[parser](self.env, self.engine, self.parser)
+        self._visitor = _parsers[parser](
+            self.env, self.engine, self.parser
+        )  # type: Optional[BaseExprVisitor]
+        assert isinstance(self._visitor, BaseExprVisitor), type(self._visitor)
```

Comment:
I don't think that the type should be Optional or that the type annotation should be added here.

adding type annotations directly to _parsers...

```
_parsers: Dict[str, Type[BaseExprVisitor]] = {
    "python": PythonExprVisitor,
    "pandas": PandasExprVisitor,
}
```
negates the need for the assert, but does mean that the type annotations added later in this PR in pandas/core/computation/pytables.py need to be revisted.